lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -688,12 +693,24 @@ monitorMetadata
 monitorMetadata tr gp DBLayer{..} = do
     settings <- atomically readSettings
     manager <- newManager defaultManagerSettings
+
+    case poolMetadataSource settings of
```

Comment:
This usually triggers when something changes the smash setting after the server started. If the server isn't healthy, we crash the worker thread.

The problem is... the caller doesn't really know, but they should! I don't think it's enough to emit warnings here, because subsequent fetch attempts will just fail forever. We probably need an API to query the status of the metadata worker.